src/e2e-test/java/io/cdap/plugin/salesforcemultiobjectsbatchsource/actions/SalesforceMultiObjectsPropertiesPageActions.java

```diff
@@ -0,0 +1,149 @@
+        ElementHelper.scrollToElement(SalesforceMultiObjectsPropertiesPage.salesforceMultiObjectsSourceInList);
+        logger.info("Click on Salesforce source from the list");
+        SalesforceMultiObjectsPropertiesPage.salesforceMultiObjectsSourceInList.click();
+    }
+
+    public static void navigateToSalesforceMultiObjectsPropertiesPage() {
+        ElementHelper.hoverOverElement(SalesforceMultiObjectsPropertiesPage.salesforceMultiObjectsSourcePluginTitle);
+        SeleniumHelper.waitElementIsVisible(
+                SalesforceMultiObjectsPropertiesPage.salesforceMultiObjectsSourcePluginPropertiesButton);
+        SalesforceMultiObjectsPropertiesPage.salesforceMultiObjectsSourcePluginPropertiesButton.click();
+    }
+
+    public static void fillReferenceName(String referenceName) {
+        logger.info("Fill Reference name: " + referenceName);
+        SalesforceMultiObjectsPropertiesPage.referenceInput.sendKeys(referenceName);
+    }
+
+    public static void fillAuthenticationProperties(String username, String password, String securityToken,
+                                                    String consumerKey, String consumerSecret) {
+        logger.info("Fill Authentication properties for user: " + username);
+        SalesforceMultiObjectsPropertiesPage.usernameInput.sendKeys(username);
+        SalesforceMultiObjectsPropertiesPage.passwordInput.sendKeys(password);
+        SalesforceMultiObjectsPropertiesPage.securityTokenInput.sendKeys(securityToken);
+        SalesforceMultiObjectsPropertiesPage.consumerKeyInput.sendKeys(consumerKey);
+        SalesforceMultiObjectsPropertiesPage.consumerSecretInput.sendKeys(consumerSecret);
+    }
+
+    public static void fillAuthenticationPropertiesForSalesforceAdminUser() {
+        SalesforceMultiObjectsPropertiesPageActions.fillAuthenticationProperties(
+                PluginUtils.getPluginPropertyValue("admin.username"),
+                PluginUtils.getPluginPropertyValue("admin.password"),
+                PluginUtils.getPluginPropertyValue("admin.security.token"),
+                PluginUtils.getPluginPropertyValue("admin.consumer.key"),
+                PluginUtils.getPluginPropertyValue("admin.consumer.secret"));
+    }
+
+    public static void fillWhiteListWithSObjectNames(String[] sObjectNames) {
+        int totalSObjects = sObjectNames.length;
+
+        if (totalSObjects > 1) {
+            for (int i = 0; i < totalSObjects - 1; i++) {
+                SalesforceMultiObjectsPropertiesPage.sObjectNameAddRowButtonsInWhiteList.get(i).click();
+            }
+        }
+
+        for (int i = 0; i < totalSObjects; i++) {
+            SalesforceMultiObjectsPropertiesPage.sObjectNameInputsInWhiteList.get(i).sendKeys(sObjectNames[i]);
+        }
+    }
+
+    public static void fillBlackListWithSObjectNames(String[] sObjectNames) {
+        int totalSObjects = sObjectNames.length;
+
+        if (totalSObjects > 1) {
```

Comment:
what will happen if totalSObjects = 1

Reply:
We need to click on the Add button only when there are more than one SObjects to be listed. 

In case of only one SObject, test will fill the default displayed fields and proceed.